src/kafka/kafka_clients.clj

```diff
@@ -0,0 +1,149 @@
+(ns kafka.kafka-clients
+  (:import (java.net InetAddress)
+           (org.apache.kafka.clients.producer KafkaProducer ProducerRecord)
+           (org.apache.kafka.common.serialization StringSerializer StringDeserializer)
+           (org.apache.kafka.clients.consumer KafkaConsumer)
+           (java.time Duration)))
+
+(def producer-config-atom
+  (atom {"value.serializer"  StringSerializer
+         "key.serializer"    StringSerializer
+         "client.id"         (.getHostName (InetAddress/getLocalHost))
+         "bootstrap.servers" "localhost:9092"
+         "acks"              "all"}))
+
+(def consumer-config-atom
+  (atom {"group.id"           "My-Group"
+         "key.deserializer"   StringDeserializer
+         "value.deserializer" StringDeserializer
+         "client.id"          (.getHostName (InetAddress/getLocalHost))
+         "bootstrap.servers"  "localhost:9092"}))
+
+(defn build-producer
+  []
+  (new KafkaProducer @producer-config-atom))
+
+(defn build-consumer
+  []
+  (let [consumer (new KafkaConsumer @consumer-config-atom)
+        _        (.subscribe consumer ["demo"])]
+    consumer))
+
+(def producer-atom (atom (build-producer)))
+
+(def consumer-atom (atom (build-consumer)))
+
+(defn setup-producer-atom-with-new-producer!
+  [new-producer]
+  (reset! producer-atom new-producer))
+
+(defn setup-consumer-atom-with-new-consumer!
+  [new-consumer]
+  (reset! consumer-atom new-consumer))
+
+(defn send-msg!
+  [record]
+  (.send @producer-atom record))
+
+(defn read-msg!
+  []
+  (let [consumer @consumer-atom
+        record-s (.poll consumer (Duration/ofMillis 100))
+        value-s  (mapv (fn [record]
+                         (str "Processed Value: " (.value record)))
```

Comment:
Why pollute the value? "Processed Value: " looks like an artifact from an old print/println.